contracts/story-nft/CachableNFT.sol

```diff
@@ -49,6 +49,26 @@ abstract contract CachableNFT is OwnableUpgradeable {
         return $.cache.length();
     }
 
+    /// @notice Returns the cache mode.
+    /// @return The cache mode, true for cache mode, false for passthrough mode.
+    function getCacheMode() external view returns (bool) {
+        return _getCacheableNFTStorage().cacheMode;
+    }
+
+    /// @notice Returns the NFT at the given index in the cache.
+    /// @param index The index of the NFT in the cache.
+    /// @return tokenId The token ID of the NFT.
+    /// @return ipId The IP ID of the NFT.
+    function getCacheAtIndex(uint256 index) external view returns (uint256 tokenId, address ipId) {
+        return _getCacheableNFTStorage().cache.at(index);
+    }
+
+    /// @notice Returns the number of NFTs in the cache.
+    /// @return The number of NFTs in the cache.
+    function getCacheLength() external view returns (uint256) {
+        return _getCacheableNFTStorage().cache.length();
+    }
```

Comment:
We have cache size function alreay.

Reply:
resolved in [1347555](https://github.com/storyprotocol/protocol-periphery-v1/pull/116/commits/13475554bc22a30fae9cb0f658eb8b4708fc22cf)